app/controllers/posts_controller.rb

```diff
@@ -18,14 +20,22 @@ def create
     @post = Post.new(post_params)
     @post.author_id = current_user.id
     if @post.save
-      flash[:success] = 'Successfully created a post!'
-      redirect_to user_posts_path(current_user)
+      redirect_to user_posts_path(current_user), notice: 'Successfully created a post!'
     else
-      flash[:error] = 'Failed to create a post.'
+      flash.now[:notice] = 'Failed to create a post.'
       render :new
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    if @post.destroy
+      redirect_to user_posts_path(@post.author_id), notice: 'Post deleted successfully'
+    else
+      redirect_to user_posts_path, notice: 'An error occurred when deleting the post'
+    end
```

Comment:
- [ ] I got an error when I wanted to delete a post that had a like and a comment. This can be solved by explicitly instructing the post associations at the model level that when a post is deleted, even its likes and comments should be deleted.  Something like this can be done: __has_many: :comments, dependent: :destroy__.

![delete](https://user-images.githubusercontent.com/53124548/231744277-91e9b153-6208-430a-b4c5-486477ddf48d.png)